gateway/ln-gateway/src/lib.rs

```diff
@@ -1612,6 +1615,75 @@ impl Gateway {
         Ok(())
     }
 
+    /// Queries the client log for payment events and returns to the user.
+    pub async fn handle_payment_log_msg(
+        &self,
+        PaymentLogPayload {
+            position,
+            limit,
```

Comment:
I'm rethinking this API. It is a bit weird to pass a global `position` and `limit` when each connected client will have a separate position in the log. @dpc any thoughts? Maybe the gateway API should take a timestamp instead and do `find_by_range` until it passes the supplied timestamp

Reply:
On second thought, I think `federation_id` should be required. Then `position` and `limit` actually make sense.